clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java

```diff
@@ -733,4 +738,18 @@ public void endOfIncrementalPushReceived(
       });
     }
   };
+
+  static ExecutionStatus getDaVinciErrorStatus(String message, Exception e) {
+    ExecutionStatus status = DVC_INGESTION_ERROR_OTHER;
+    if (e instanceof MemoryLimitExhaustedException) {
+      status = DVC_INGESTION_ERROR_MEMORY_LIMIT_REACHED;
+    } else if (e instanceof VeniceException) {
+      if (message.startsWith("Disk is full")) {
```

Comment:
Is it possible to have a specific exception like `DiskFullException` instead of relying on the message?

Reply:
Good call. Added `DiskLimitExhaustedException`.